webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/Authorizations.scala

```diff
@@ -433,6 +459,30 @@ object RudderRoles {
     } yield ()
   }
 
+  /*
+   * Register built-in roles. These roles are typically provided by plugins and can't be unloaded or
+   * changed at run-time, contrary to user-provided custom roles.
+   * If the registered role is a new name, then it is just added to the list of built-in roles.
+   * If the registered role has an existing name, then it *replaces* the existing built-in roles. This is typically
+   * done by a plugin to extends an existing core built-in role with new permissions.
+   * Administrator and NoRights roles are special and can't be overridden.
+   */
+  def registerBuiltin(roles: List[Role]): IOResult[Unit] = {
```

Comment:
I think this function should extend the existing instead of replacing. this would allow getting the existing roles in plugins, you just define your plugin roles, with authorization provided by the plugin (and only them, you don't need too get existing ones)